### DISCUSSION TITLE
Storyboards, videos, and storyboard sound samples do not sync with the music with an adjusted offset

### DISCUSSION THREAD

**Discussion by Walavouchey** [ppy/osu#16256] [Q&A] [answered]
This is the case on both stable and lazer, but it would be nice to fix in lazer. Any beatmap that uses storyboards or videos in any way that should sync to the music will sound off by any offset change (global or local, although local doesn't exist in lazer).

This is a problem for the common usage of using universal offset to sync keytaps to the music (especially for mania players as has become apparent by the [RC hitsound requirement debate](https://osu.ppy.sh/community/forums/topics/1418542?n=47)), although playing with storyboards is unfortunately an unpopular playstyle.

Edit: Since the track doesn't skip forward or distort when changing offset, it means that samples are affected and not the track. With a positive universal offset, sb samples shift back (play early).

https://user-images.githubusercontent.com/36758269/147415991-cc051c7c-71b2-4ef5-b78e-8826dcf1b7e8.mp4

[metronome test map (rename to .osz)](https://github.com/ppy/osu/files/7777342/sb-metronome.zip)

**Comment by bdach** [comment]
i think in this case putting the storyboard under the master gameplay clock is wrong, because the offset is supposed to affect user input rather than storyboard samples, and as such storyboard should be using the raw track clock, but i'm not 100% on this.

**Reply by peppy** [reply]
Actually the opposite.

**Comment by peppy** [comment, marked as answer]
This is correct behaviour. If a user is using the universal offset to account for input latency, they will probably want to mute samples.

The offset is made to sync samples to music.

**Reply by Walavouchey** [reply]
Storyboard sound samples are always synced to the video, beatmap and storyboard (they do not react to keypresses), so I guess they were never designed to sync with the music in the first place. I suppose audio file editing is the only best option for passive hitsounds, and using SB samples should only be done for sounds that are okay to play out of sync (since sound samples can be played on the pass or fail layers). I'll leave it at that.

Muting samples can only be done by muting all sound effects (or by not playing with storyboard at all), so I suppose syncing keytaps with the music and playing with storyboards will always be a trade-off.

**Reply by peppy** [reply]
I think you are misunderstanding. Or this issue has been worded incorrectly.

Adjusting UO should allow you to synchronise storyboard sounds, and autoplay hitsounds, to sound 100% in time. If that's the case, then there's no issue. If that's not the case, then there's an issue.

**Reply by Walavouchey** [reply]
See my response on https://github.com/ppy/osu/discussions/16252. I understand now that UO is not designed to account for input latency, but that reveals a bigger issue.

**Reply by vernonlim** [reply]
Could there not just be an option for storyboards to not move with UO? I'm pretty sure it would solve the main problem here (storyboarded samples being off-time when using UO for keytap timing) without breaking hitsound timing being the normal paradigm.

**Reply by Walavouchey** [reply]
The issue is precisely that there isn't a dedicated offset for judgements. If such a thing existed, none of what I've said would be an issue, since the core problem is using UO for keytap timing, which it wasn't designed for.
